benchmarks/benchmarks_common.h

```diff
@@ -0,0 +1,101 @@
+#include <sstream>
+#include <string>
+#include <vector>
+
+// We use this instead of rand to ensure it's xplat
+int pseudo_random(int seed) {
+  constexpr uint64_t CONSTANT_A = 0xeece66d5deece66dULL;
+  constexpr uint64_t CONSTANT_C = 2147483647;
+
+  uint64_t val = CONSTANT_A * seed + CONSTANT_C;
+  return (int)(val & 0xFFFFFFFF);
+}
+
+class prng {
+  int val;
+
+public:
+  prng(int initial_seed) : val(pseudo_random(initial_seed)) {}
+  uint32_t next_uint() {
+    uint32_t res = val & 0x7FFFFFFF;
+    val = pseudo_random(val);
+    return res;
+  }
+};
+
+class cb_decision_gen {
+  int shared_features, action_features, actions_per_decision;
+  std::vector<std::string> actions_set;
+  prng rand;
+  std::string temp_str;
+  bool passthrough;
+
+  std::string mk_feature_vector(int count, uint32_t max_idx) {
+    std::stringstream str;
+    str << "{";
+    std::set<uint32_t> added_idx;
+    int added = 0;
+    while (added < count) {
+      auto idx = rand.next_uint() % max_idx;
+      if (added_idx.find(idx) == added_idx.end()) {
+        if (added > 0)
+          str << ",";
+        str << "\"_" << idx << "_f\":1";
+
+        ++added;
+        added_idx.insert(idx);
+      }
+    }
+    str << "}";
+    return str.str();
+  }
+
+public:
+  cb_decision_gen(int shared_features, int action_features,
```

Comment:
In general, only templates or small inline methods should be in headers.

Reply:
true

Reply:
split files